Update models selection

bishwobhandari opened the pull request:
closes #99 updated models selection functions
- removed all models constants
-prior weights are by enabling models only
- added default models when dataset type changes and new session created.

shapiromatron commented on frontend/src/components/Main/ModelsForm/ModelsCheckBox.js:
```diff
@@ -1,73 +1,734 @@
+                    <tr>
+                        <td className="text-left align-middle">Exponential</td>
+                        <td>
+                            <input
```
so much duplication here, make a new component that returns the td we want. The goal here is to reduce the # lines of code

```javascript
<ModelCheckboxTd type="frequentist_restricted" model="Exponential"/>
```

shapiromatron commented on frontend/src/components/Main/ModelsForm/ModelsCheckBox.js:
```diff
@@ -1,73 +1,734 @@
+                                checked={isChecked("bayesian_model_average", "Logistic")}></input>
+                        </td>
+                        <td>
+                            <input
```
again, make a reusable component `<PriorWeightInputTd/>`

shapiromatron commented on frontend/src/components/Main/ModelsForm/ModelsCheckBoxHeader.js:
```diff
@@ -1,40 +1,88 @@
+            <tr>
+                <th>Models</th>
+                <th>
+                    <input
```
make a reusable component here too. For all these reusable components, I just make it at the top of this file since it' sonly used in one place.

```javascript
const SelectAllComponent = observer((props)=>{
    return <th>....</th>
})


<SelectAllComponent store={store} ... />



```

shapiromatron commented on frontend/src/components/Main/ModelsForm/ModelsReadOnly.js:
```diff
@@ -1,40 +1,197 @@
+                <tbody>
+                    <tr>
+                        <td className="text-left align-middle">Exponential</td>
+                        <td>
```
again, make a helper function here... too much redundancy

shapiromatron reviewed (changes requested):
Let's build some reusable components to get the duplicative code down, we should get a net reduction of LOC from this PR.

shapiromatron commented:
> ready for review @shapiromatron

Done! Let's get the LOC down, then I can review the store logic after that's done...

bishwobhandari commented:
@shapiromatron  ready for review again. I have made changes. created reusable components for td
-modelsReadOnly page is deleted, thought it might be just easier to work in modelsCheckbox with can Edit value.

shapiromatron reviewed (changes requested):
@bishwobhandari Made some changes to code to improve readability and get line count down.

I noticed a few bugs however:

1. When creating a "new" run, no default models are checked, and no default option set exists. I have to change the model type to get the default models
2. After changing the model type so that default models are checked and adding a dataset, the " At least one model is selected" check still has an X - this check appears to always fail?

bishwobhandari commented:
@shapiromatron ready for review. 
- updated component with componentDidMount() hook to get default models and options selected. 
- updated main store to get the models from models store so that check can pass.

shapiromatron reviewed (approved):
Looks good! Made a few additional changes; 1) renamed Quantal Linear; 2) fixed logic for setting defaults
